File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-7.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        l = len(arr)
        rightMaxSoFar = -1

        for i in range(l - 1, -1, -1):
            val = arr[i]
            arr[i] = rightMaxSoFar
            rightMaxSoFar = max(val, rightMaxSoFar)
        
        return arr

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-7.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, maxP = 0;

        for (int r = 1; r < prices.size(); ++r) {
            if (prices[r] > prices[l]) {
                maxP = max(maxP, prices[r] - prices[l]);
            }
            else {
                l = r;
            }
        }

        return maxP;
    }
};
